esphome/components/mipi_spi/display.py

```diff
@@ -0,0 +1,474 @@
+import logging
+
+from esphome import pins
+import esphome.codegen as cg
+from esphome.components import display, spi
+from esphome.components.spi import TYPE_OCTAL, TYPE_QUAD, TYPE_SINGLE
+import esphome.config_validation as cv
+from esphome.config_validation import ALLOW_EXTRA
+from esphome.const import (
+    CONF_BRIGHTNESS,
+    CONF_COLOR_ORDER,
+    CONF_CS_PIN,
+    CONF_DATA_RATE,
+    CONF_DC_PIN,
+    CONF_DIMENSIONS,
+    CONF_ENABLE_PIN,
+    CONF_HEIGHT,
+    CONF_ID,
+    CONF_INIT_SEQUENCE,
+    CONF_INVERT_COLORS,
+    CONF_LAMBDA,
+    CONF_MIRROR_X,
+    CONF_MIRROR_Y,
+    CONF_MODEL,
+    CONF_OFFSET_HEIGHT,
+    CONF_OFFSET_WIDTH,
+    CONF_RESET_PIN,
+    CONF_ROTATION,
+    CONF_SWAP_XY,
+    CONF_TRANSFORM,
+    CONF_WIDTH,
+)
+from esphome.core import TimePeriod
+
+from ..const import CONF_DRAW_ROUNDING
+from ..lvgl.defines import CONF_COLOR_DEPTH
+from . import (
+    CONF_BUS_MODE,
+    CONF_DRAW_FROM_ORIGIN,
+    CONF_NATIVE_HEIGHT,
+    CONF_NATIVE_WIDTH,
+    CONF_PIXEL_MODE,
+    CONF_SPI_16,
+    CONF_USE_AXIS_FLIPS,
+    DOMAIN,
+    MODE_BGR,
+    MODE_RGB,
+)
+from .models import (
+    DELAY_FLAG,
+    MADCTL_BGR,
+    MADCTL_MV,
+    MADCTL_MX,
+    MADCTL_MY,
+    MADCTL_XFLIP,
+    MADCTL_YFLIP,
+    DriverChip,
+    amoled,
+    cyd,
+    ili,
+    jc,
+    lanbon,
+    lilygo,
+    waveshare,
+)
+from .models.commands import BRIGHTNESS, DISPON, INVOFF, INVON, MADCTL, PIXFMT, SLPOUT
+
+DEPENDENCIES = ["spi"]
+
+LOGGER = logging.getLogger(DOMAIN)
+mipi_spi_ns = cg.esphome_ns.namespace("mipi_spi")
+MIPI_SPI = mipi_spi_ns.class_(
```

Comment:
We try to have the variables in python match exact to c++ where possible
```suggestion
MipiSpi = mipi_spi_ns.class_(
```